### ISSUE TITLE
Breaking change warnings in connection overview page 

### ISSUE THREAD

**Issue by erohmensing** [airbytehq/airbyte#30945]
## What to do 

On the connections overview page, we currently have a method of showing warning badges: 

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBbzN4IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--bc3000074b820ff23a8cfda749473d054a54163a/image.png)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBby94IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--c8376373447bfec09c9a49dd36139c0ef5b96333/image.png)

We want breaking changes added to this warning, similar to how they were added to the banners on the connection page next to those schema change banners. 

We already have a red variant of this that we show when schema changes are breaking. I think this means we should:
* combine elements that will go into the tooltip: 
  * Existing: 
    * "Breaking schema updates detected"
    * "Non-breaking schema updates detected"
   * New: 
    * "Source version upgrade required" 
    * "Destination version upgrade required" 
    * "Source upgrade deadline passed"
    * "Destination upgrade deadline passed"
 
Alternatively we could not distinguish between sources/destinations here (and risk having to show both) - we could just have 
"Connector version upgrade required" and
"Connector upgrade deadline passed" 

With regards to combining multiple possible warnings - anything that is "red" (breaking schema changes, deadline passed) would make the icon red instead of yellow, even if there are yellow warnings as well. 

I'd recommend that we just stack them vertically within the tooltip. With the initial suggestion you could have 3 stacked vertically, with the replacement, only 2. 

This copy is meant to resemble the notification settings

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcTd4IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--559c343579acdbf911ffaea2c696bd36d819ed93/image.png)

## (WIP) Suggested implementation: 
* Add this info to `web_backend/connections/list`. this is a frontend-only endpoint, so we should be able to mess with the format. 
Current format: 
```
        {
            "connectionId": "01141f05-7c7c-44ec-8f3c-a5eef4810dad",
            "name": "Facebook Marketing <> End-to-End Testing (/dev/null)",
            "scheduleType": "manual",
            "status": "active",
            "source": {
                "sourceId": "7248a605-33b4-4f27-ac9c-a0b3300185cd",
                "name": "Facebook Marketing",
                "sourceDefinitionId": "e7778cfc-e97c-4458-9ecb-b4f2bba8946c",
                "sourceName": "Facebook Marketing",
                "icon": ...
            },
            "destination": {
                "destinationId": "98b4b368-ee35-4b04-8201-cd24ea815376",
                "name": "End-to-End Testing (/dev/null)",
                "destinationDefinitionId": "a7bcc9d8-13b3-4e49-b80d-d020b90045e3",
                "destinationName": "End-to-End Testing (/dev/null)",
                "icon": ...
            },
            "latestSyncJobCreatedAt": 1689148182,
            "latestSyncJobStatus": "succeeded",
            "isSyncing": false,
            "schemaChange": "non_breaking"
        },
```
Suggestion: change the schemaChange property to be a nested prop that refers to connectionWarnings or something of the sort. 
```
"warnings" : {
  "schemaChange": "non_breaking",
  "breakingChanges: {
      "source": $ref: ActorDefinitionVersionBreakingChanges,
      "destination": $ref: ActorDefinitionVersionBreakingChanges
   }
}
```
As an alternative, we could add breaking changes to the `SourceSnippetRead` and `DestinationSnippetRead` and read off of those. This would be a less disruptive change, but would spread the information required to render the badge over multiple places. 

## Important context 
We also want this information to be visible on the sources/destinations page. I'm imagining it looking very similar to the

**Comment by nora-airbyte** [comment]
Agreed! I think we should stack messages in the tooltip when there's more than one, but change the indicator to a count. I also added icons to the messages within the tooltip to differentiate them:

![Stacked_messges](https://github.com/airbytehq/airbyte/assets/146028830/ac933f5a-eb2b-48e2-a004-0e96cff4c81f)

Also agreed that the count / indicator takes the color of the most severe message in the stack.

**Comment by evantahler** [comment]
Grooming: moving to icebox for now
